charts/kong/templates/_helpers.tpl

```diff
@@ -1710,3 +1711,15 @@ extensions/v1beta1
 {{- end -}}
 {{- (toYaml $proxyReadiness) -}}
 {{- end -}}
+
+{{- define "kong.router_flavor" -}}
+{{- if .Values.env.router_flavor -}}
+.Values.env.router_flavor
+{{- else -}}
+    {{- if (and  .Values.ingressController.enabled  (semverCompare ">= 3.0.0" (include "kong.effectiveVersion" .Values.ingressController.image))) -}}
+        expressions
+    {{- else -}}
+        traditional_compatible
```

Comment:
As I understand this also changes the default when KIC is < 3.0.0 from `traditional` to `traditional_compatible`, is this intended?